studio/lib/queries/pages.ts

```diff
@@ -1,87 +1,105 @@
 import { groq } from "next-sanity";
 
-// TODO: set all sectons.ref's to slug.current
+export const LANGUAGE_FIELD_FRAGMENT = groq`
+  "language": $language
+`;
+
+export function translatedFieldFragment(fieldName: string) {
+  return groq`${fieldName}[_key == $language][0].value`;
```

Comment:
Why have you chosen this approach instead of: 

```
  // "basicTitle": basicTitle[$language],
  // "slug": slug[$language].current,
 ```

Reply:
I think, if possible, the suggestion above would give us less bloat and a cleaner approach.

Reply:
Because that's the query required by the shape of the data: https://github.com/sanity-io/sanity-plugin-internationalized-array/blob/main/README.md#shape-of-stored-data

Reply:
Ah my bad, I was looking ad doc translation 💡 

I am still concerned however with the amount of bloat and complexity this approach causes. I'd vote for opting out of the added fragments and if absolutely needed we could go back and revisit.

Reply:
The whole point of using fragments is to reduce bloat and address the complexity introduced in queries when using translated strings. If we don't use fragments, we have to repeat the same GROQ expressions in all the places we have translations. This increases the chance that we make mistakes when writing queries.

Reply:
I agree, I just think that for cases where there's only one line, such as `basicTitle[_key == $language][0].value` there's no need for fragments

Reply:
Let me clarify further. The Link fragments are absolutely ok! 
But I think the `translatedFieldFragment()` is redundant

Reply:
Ok, good 🤗 

Based on your initial comment, I think we can agree that `[_key == $language][0].value` is not very readable or clean. It's also used many places, so reusing the expression makes sense. `translatedFieldFragment()´ is therefore both an attempt at demystifying the expression and to enable reuse.

Reply:
No, that's not what I meant. 

I meant that `[_key == $language][0].value` is sufficient enough and to omit the use of `translatedFieldFragment()`

Reply:
I don't agree that it's redundant, seeing as we already use it nine different places with just these initial translations. However, I'm fine with removing it if it means we can get this merged.

Reply:
I will leave this as is for now. We can always inline everything later if it becomes an issue.